functions/source/CognitoUserInit/cfn_cognito_init.py

```diff
@@ -0,0 +1,64 @@
+from operator import length_hint
+import cfnresponse
+import boto3
+import botocore
+import json
+import random
+import string
+import secrets
+import logger
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+def rand_seq(seq, at_least: int, at_most: int) -> list:
+    n = secrets.choice(range(at_least, at_most + 1))
+    return [secrets.choice(seq) for _ in range(n)]
+
+def lambda_handler(event, context):
+    """Lambda Handler for dealing with creating a new cognito user using AWS Cognito
+
+    Args:
+        event (dict): Event dictionary from custom resource
+        context (obj): Context manager
+    """
+
+    user_pool_id = event['ResourceProperties']['UserPoolId']
+    username = 'admin'
+    
+    # Generate random word using secrets for 20 characters long
+    length = 20
+    password = []
+    password += rand_seq(string.punctuation, 1, 3)
+    password += rand_seq(string.digits, 1, 3)
+    password += rand_seq(string.ascii_uppercase, 3, 9)
+    password = length - len(password)
+    password += rand_seq(string.ascii_lowercase, n, n)
+    password = "".join(password)
+    
+
+    if event['RequestType'] in ['Create', 'Delete', 'Update']:      
+        # Run in the cloud to make cognito user
+        client = boto3.client('cognito-idp') 
+
+        try:
+            cidp_response = client.admin_create_user(
+                UserPoolId = user_pool_id,
+                Username = username,
+                TemporaryPassword = password
+            )
+
+            cfn_response_from_cidp = json.dumps(cidp_response, indent=4, sort_keys=True, default=str)
+            cfn_response_from_cidp = json.loads(cfn_response_from_cidp)
+            cfn_response_from_cidp['TemporaryPassword'] = password
+
+            # Send response back with CIDP Response to CFN
+            cfnresponse.send(event, context, cfnresponse.SUCCESS, cfn_response_from_cidp)
```

Comment:
Please keep the response sent back to cfn light and clean; e.g.
```python
response_data = {}
response_data["Username"] = username
response_data["TemporaryPassword"] = password
cfnresponse.send(..., response_data)
```

Reply:
Made the change - I thought we'd want to send the response that came back from AWS Cognito Client. It was quite verbose though so good point :)